core/audits/preload-lcp-image.js

```diff
@@ -116,6 +110,29 @@ class PreloadLCPImageAudit extends Audit {
     };
   }
 
+  /**
+   * Match the LCP event with the paint event to get the URL of the image actually painted.
+   * This could differ from the `ImageElement` associated with the nodeId if e.g. the LCP
+   * was a pseudo-element associated with a node containing a smaller background-image.
+   * @param {LH.Trace} trace
+   * @param {LH.Artifacts.ProcessedNavigation} processedNavigation
+   * @return {string | undefined}
+   */
+  static getLcpUrl(trace, processedNavigation) {
```

Comment:
Will we be needing this logic elsewhere? maybe trace processor could do it?

Reply:
that could make sense, or in `trace-elements` maybe? But maybe we should wait for the need (e.g. if we want to do something with #11629, the URL is included on the CDP event, so we won't need it there)